internal/command/command.go

```diff
@@ -182,3 +182,49 @@ func exists(ctx context.Context, filter preparation.FilterToQueryReducer, wm exi
 	}
 	return wm.Exists(), nil
 }
+
+func (c *Commands) processWithLast(ctx context.Context, validation preparation.Validation) (*domain.ObjectDetails, error) {
+	cmds, err := preparation.PrepareCommands(ctx, c.eventstore.Filter, validation)
+	if err != nil {
+		return nil, err
+	}
+	events, err := c.eventstore.Push(ctx, cmds...)
+	if err != nil {
+		return nil, err
+	}
+	return &domain.ObjectDetails{
+		Sequence:      events[len(events)-1].Sequence(),
+		EventDate:     events[len(events)-1].CreationDate(),
+		ResourceOwner: events[len(events)-1].Aggregate().ResourceOwner,
+	}, nil
+}
+func (c *Commands) processWithFirst(ctx context.Context, validation preparation.Validation) (*domain.ObjectDetails, error) {
+	cmds, err := preparation.PrepareCommands(ctx, c.eventstore.Filter, validation)
+	if err != nil {
+		return nil, err
+	}
+	events, err := c.eventstore.Push(ctx, cmds...)
+	if err != nil {
+		return nil, err
+	}
+	return &domain.ObjectDetails{
+		Sequence:      events[0].Sequence(),
+		EventDate:     events[0].CreationDate(),
+		ResourceOwner: events[0].Aggregate().ResourceOwner,
+	}, nil
+}
+func (c *Commands) processWithID(ctx context.Context, validation preparation.Validation) (string, *domain.ObjectDetails, error) {
+	cmds, err := preparation.PrepareCommands(ctx, c.eventstore.Filter, validation)
+	if err != nil {
+		return "", nil, err
+	}
+	events, err := c.eventstore.Push(ctx, cmds...)
+	if err != nil {
+		return "", nil, err
+	}
+	return events[len(events)-1].Aggregate().ID, &domain.ObjectDetails{
+		Sequence:      events[len(events)-1].Sequence(),
+		EventDate:     events[len(events)-1].CreationDate(),
+		ResourceOwner: events[len(events)-1].Aggregate().ResourceOwner,
+	}, nil
```

Comment:
i see the point in having a function, which will return the object details
and as the order of events is relevant, it depends where to take the information from, but it looks arbitrary to use last or first... and withID will always use the last
can we just use the `pushAppendAndReduce` and extract the info from the writeModel afterwards?

Reply:
could do that, then all the prepare functions have to be changed to have a writemodel as parameter, how exactly should be define the structure now?